src/qr.rs

```diff
@@ -649,6 +683,20 @@ pub async fn set_config_from_qr(context: &Context, qr: &str) -> Result<()> {
                 .set_config_internal(Config::WebrtcInstance, Some(&instance_pattern))
                 .await?;
         }
+        Qr::Socks5Proxy { host, port } => {
+            // disable proxy before changing settings to not use a combination of old and new
+            context
+                .set_config_bool(Config::Socks5Enabled, false)
+                .await?;
+
+            context.set_config(Config::Socks5Host, Some(&host)).await?;
+            context
+                .set_config_u32(Config::Socks5Port, u32::from(port))
+                .await?;
+            context.set_config(Config::Socks5User, None).await?;
+            context.set_config(Config::Socks5Password, None).await?;
+            context.set_config_bool(Config::Socks5Enabled, true).await?;
```

Comment:
I meant the following "solution" for now, to not complicate the code with transactions or so. Anyway, this is minor.
```suggestion
            // invalidate proxy before changing settings to not use a combination of old and new
            context.set_config(Config::Socks5Host, None).await?;

            context
                .set_config_u32(Config::Socks5Port, u32::from(port))
                .await?;
            context.set_config(Config::Socks5User, None).await?;
            context.set_config(Config::Socks5Password, None).await?;
            context.set_config(Config::Socks5Host, Some(&host)).await?;
            context.set_config_bool(Config::Socks5Enabled, true).await?;
```